x-pack/plugins/apm/server/routes/settings/agent_configuration.ts

```diff
@@ -95,10 +96,23 @@ const deleteAgentConfigurationRoute = createApmServerRoute({
       `Deleting config ${service.name}/${service.environment} (${config._id})`
     );
 
-    return await deleteConfiguration({
+    const deleteConfigurationResult = await deleteConfiguration({
       configurationId: config._id,
       setup,
     });
+
+    if (resources.plugins.fleet) {
+      await syncAgentConfigsToApmPackagePolicies({
+        core,
+        fleetPluginStart: await resources.plugins.fleet.start(),
+        setup,
+      });
+      logger.info(
+        `Saved latest agent settings to Fleet integration policy for APM.`
```

Comment:
Since you're logging something, maybe would be good to have a specific message when it is deleting?